### PR TITLE
[Enabler] [zos_mvs_raw] Changes to ensure tests portability

### LINKED ISSUE #1590: [Enabler][Portability] [zos_mvs_raw] Ensure test suite portability by removing hard coded data.
### Is there an existing issue for this?

- [X] There are no existing issues.

### Enabler description
This issue is tracked by epic: [[Epic][Portability] Ensure test suite execution portability & parallelism.](https://github.com/ansible-collections/ibm_zos_core/issues/1585)

#### Issue:
In order to ensure test portability and parallel execution we need to remove data set names and user names that are hardcoded.
In this test the following needs to be removed:
- [ ] Use of USER.PRIVATE.PROCLIB.

### Ansible module

zos_mvs_raw

### PR DESCRIPTION
**Pull request by rexemin**
##### SUMMARY
Fixes #1590.

##### ISSUE TYPE

- Enabler Pull Request

##### COMPONENT NAME
zos_mvs_raw

##### ADDITIONAL INFORMATION
Shortened the IBM-1047-encoded string used in a test (`test_input_return_text_content_encodings`). Dataset names used for IDCAMS are now randomly generated and this test had its expected return hardcoded to the dataset name too.

### CODE REVIEW

**Comment by rexemin** [comment]
Screenshot 2024-08-01 at 14 58 04
Screenshot 2024-08-01 at 14 58 23

**Review by richp405 — APPROVED** [review]

File: tests/functional/modules/test_zos_mvs_raw_func.py [hunk 1]
```diff
@@ -31,6 +31,16 @@
 IDCAMS_INVALID_STDIN = " hello world #$!@%!#$!@``~~^$*%"
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_mvs_raw_func.py:32]: it looks like all the uses of the `IDCAMS_STDIN` variable have been removed, so it's probably a good idea to remove the variable definition too (line 30)

File: tests/functional/modules/test_zos_mvs_raw_func.py [hunk 2]
```diff
@@ -55,6 +65,8 @@ def test_disposition_new(ansible_zos_module):
     try:
         hosts = ansible_zos_module
         default_data_set = get_tmp_ds_name()
+        idcams_dataset, idcams_args = get_temp_idcams_dataset(hosts)
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_mvs_raw_func.py:68]: not a big deal, but perhaps there is a more descriptive name for `idcams_args`, maybe something like `idcams_listcat_cmd` or `idcams_listcat_dataset_cmd` or `listcat_idcams_dataset_cmd`?

```python
      "dd_input":{
          "dd_name":SYSIN_DD,
          "content": listcat_idcams_dataset_cmd
      }
```
Even out of context, i'd have some idea of what the "content" section is and what this test case is doing

File: tests/functional/modules/test_zos_mvs_raw_func.py [hunk 3]
```diff
@@ -123,57 +138,63 @@ def test_dispositions_for_existing_data_set(ansible_zos_module, disposition):
...
+                    }
+                },
+            ],
+        )
+        for result in results.contacted.values():
+            assert result.get("ret_code", {}).get("code", -1) == 0
+            assert len(result.get("dd_names", [])) > 0
+            for backup in result.get("backups"):
+                backup.get("backup_name")[:6] == tmphlq
+        results = hosts.all.zos_data_set(name=default_data_set, state="absent")
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_mvs_raw_func.py:193]: a nit-pick here, i know you didnt change this, but for the sake of consistency, can this line deleting `default_data_set` be moved into the `finally` block?

File: tests/functional/modules/test_zos_mvs_raw_func.py [hunk 4]
```diff
@@ -1992,10 +2103,23 @@ def test_concatenation_fail_with_unsupported_dd_type(ansible_zos_module):
...
         hosts = ansible_zos_module
+
         default_data_set = "ANSIBLE.USER.PRIVATE.TEST"
         hosts.all.zos_data_set(name=default_data_set, state="present", type="seq")
+
         hosts.all.file(path=DEFAULT_PATH, state="directory")
         hosts.all.file(path=DEFAULT_PATH_WITH_FILE, state="absent")
+
+        idcams_dataset, idcams_args = get_temp_idcams_dataset(hosts)
+        dds.append(
```
Comment (ketankelkar) [inline tests/functional/modules/test_zos_mvs_raw_func.py:2114]: clever!!

**Review by ketankelkar — APPROVED** [review]
Overall, LGTM. I had a few suggested changes, but nothing critical/required, hence the approval

**Pull request merged**